src/node.rs

```diff
@@ -301,36 +328,414 @@ impl Taffy {
+
+        assert_eq!(taffy.child_count(node).unwrap(), 2);
+
+        taffy.remove_child_at_index(node, 0).unwrap();
+        assert_eq!(taffy.child_count(node).unwrap(), 1);
+        assert_eq!(taffy.children(node).unwrap()[0], child1);
+
+        taffy.remove_child_at_index(node, 0).unwrap();
+        assert_eq!(taffy.child_count(node).unwrap(), 0);
+    }
+
+    #[test]
+    fn replace_child_at_index() {
+        let mut taffy = Taffy::new();
+
+        let child0 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let child1 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+
+        let node = taffy.new_with_children(FlexboxLayout::default(), &[child0]).unwrap();
+        assert_eq!(taffy.child_count(node).unwrap(), 1);
+        assert_eq!(taffy.children(node).unwrap()[0], child0);
+
+        taffy.replace_child_at_index(node, 0, child1).unwrap();
+        assert_eq!(taffy.child_count(node).unwrap(), 1);
+        assert_eq!(taffy.children(node).unwrap()[0], child1);
+    }
+    #[test]
+    fn test_child_at_index() {
+        let mut taffy = Taffy::new();
+        let child0 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let child1 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let child2 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let node = taffy.new_with_children(FlexboxLayout::default(), &[child0, child1, child2]).unwrap();
+
+        assert!(if let Ok(result) = taffy.child_at_index(node, 0) { result == child0 } else { false });
+        assert!(if let Ok(result) = taffy.child_at_index(node, 1) { result == child1 } else { false });
+        assert!(if let Ok(result) = taffy.child_at_index(node, 2) { result == child2 } else { false });
+    }
+    #[test]
+    fn test_child_count() {
+        let mut taffy = Taffy::new();
+        let child0 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let child1 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let node = taffy.new_with_children(FlexboxLayout::default(), &[child0, child1]).unwrap();
+
+        assert!(if let Ok(count) = taffy.child_count(node) { count == 2 } else { false });
+        assert!(if let Ok(count) = taffy.child_count(child0) { count == 0 } else { false });
+        assert!(if let Ok(count) = taffy.child_count(child1) { count == 0 } else { false });
+    }
+    #[test]
+    fn test_children() {
+        let mut taffy = Taffy::new();
+        let child0 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let child1 = taffy.new_with_children(FlexboxLayout::default(), &[]).unwrap();
+        let node = taffy.new_with_children(FlexboxLayout::default(), &[child0, child1]).unwrap();
+
+        let children = taffy.children(node).unwrap();
+        assert_eq!(children, vec![child0, child1]);
```

Comment:
This will need to be changed to be a `sys::Vec` instead of `std::vec::Vec`

Reply:
Fixed by manually pushing to the vector instead of using the macro to create it.